tests/e2e/kafka.go

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// requireDeleteKafkaTopic deletes kafkaTopic resource.
+func requireDeleteKafkaTopic(kubectlOptions k8s.KubectlOptions, topicName string) {
+	It("Deleting KafkaTopic CR", func() {
+		deleteK8sResource(kubectlOptions, defaultDeletionTimeout, kafkaTopicKind, "", topicName)
+	})
+}
+
+// requireDeployingKafkaTopic deploys a kafkaTopic resource from a template
+func requireDeployingKafkaTopic(kubectlOptions k8s.KubectlOptions, topicName string) {
+	It("Deploying KafkaTopic CR", func() {
+		applyK8sResourceFromTemplate(kubectlOptions,
+			kafkaTopicTemplate,
+			map[string]interface{}{
+				"Name":      topicName,
+				"TopicName": topicName,
+				"Namespace": kubectlOptions.Namespace,
+			},
+		)
+		err := waitK8sResourceCondition(kubectlOptions, kafkaTopicKind, "jsonpath={.status.state}=created", defaultTopicCreationWaitTime, "", topicName)
+		Expect(err).ShouldNot(HaveOccurred())
```

Comment:
**Note**: According to https://pkg.go.dev/github.com/onsi/gomega#Assertion typically expect uses `NotTo` compared to `ShouldNot`, but I'm not picky about that, but we should use one of the two throughout the tests for easier human pattern recognition. I don't mind which one we use, if we feel like ShouldNot is easier to read, fine for me. I used NotTo due to being used to that, but no hard feelings.

Reply:
I saw somewhere that the "Expect(err).ShouldNot(HaveOccurred())" should be used to check error

Reply:
Could you search around just quickly in 5m what is the official suggestion here?
What I linked is the official gomega docs, so I don't understand why some tooltip would be the direct opposite.
But as I said, I don't care which, whatever is preferred officially is fine for me, I care about not mixing the two.